src/app/qgssnappingwidget.cpp

```diff
@@ -225,14 +235,18 @@ QgsSnappingWidget::QgsSnappingWidget( QgsProject* project, QgsMapCanvas* canvas,
   modeChanged();
   updateToleranceDecimals();
 
-  restoreGeometry( QSettings().value( "/Windows/SnappingWidget/geometry" ).toByteArray() );
+  mSettings = new QSettings();
```

Comment:
QSettings is fine to be constructed and thrown away, no need to keep it as a class member - and we do not need to mess with pointers and memory leaks ;-)